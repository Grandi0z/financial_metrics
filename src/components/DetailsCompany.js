import React from 'react';
import Container from 'react-bootstrap/Container';
import Card from 'react-bootstrap/Card';
import PropTypes from 'prop-types';
import { v4 as uuidv4 } from 'uuid';
import 'bootstrap/dist/css/bootstrap.min.css';
import { Link } from 'react-router-dom';
import styles from '../styles/details.module.css';
import {
  splitTitle,
  splitDate,
  capitalizeAndSpace,
  filterMetricsValues,
  setBackGround,
} from '../utils/filtersFunctions';
import {
  myBackground1,
  myBackground2,
  companyTitleBackground,
} from '../utils/customizedObject';

const DetailsCompany = (props) => {
  const { company, profile } = props;
  const { companyName, fullTimeEmployees, image } = profile;

  let backgroundCheck = false;

  const titleGroung = setBackGround(companyTitleBackground, image);
  // take the key (property) of company
  let properties;
  // contain only metrics values remove string values
  let newCompany;
  if (company) {
    newCompany = filterMetricsValues(company);
    properties = Object.keys(newCompany);
  }

  return (
    <Container style={{ padding: 0 }}>
      {newCompany && (
        <>
          <div className="bar">
            <div>
              <Link to="/">
                <i className="bi bi-chevron-left" />
              </Link>
            </div>
            <span>Details</span>
            <div className="navLinks">
              <Link href="/#">
                <i className="bi bi-mic-fill" />
              </Link>
              <Link href="/#">
                <i className="bi bi-gear-wide" />
              </Link>
            </div>
          </div>
          <div className="bar-desk">
            <div>
              <Link to="/" style={{ textDecoration: 'none' }}>
                Companies
              </Link>
            </div>
            <span>most Companies</span>
            <div className="navLinks">
              <Link href="/#" style={{ textDecoration: 'none' }}>
                <i className="bi bi-gear-wide" />
                Settings
              </Link>
            </div>
          </div>
          <header className="companyTitle" style={titleGroung}>
            <h2 className={styles.titleDetails}>
              {companyName}
              <span className={styles.employees}>
                {fullTimeEmployees}
                {' '}
                Employees
              </span>
            </h2>
            <h4 className={styles.subTitleDetails}>
              {splitTitle(companyName)}
              &apos;s metrics -
              {' '}
              {splitDate(company.date)}
            </h4>
          </header>
          <ul className={styles.ulDetails}>
            {properties.map((property) => {
              backgroundCheck = !backgroundCheck;
              return (
                <Card key={uuidv4()} className={styles.card}>
                  <Card.Body
                    style={backgroundCheck ? myBackground1 : myBackground2}
                    className={styles.liDetails}
                  >
                    <span>{capitalizeAndSpace(property)}</span>
                    <span className={styles.propertyValue}>
                      {' '}
                      {newCompany[property]}
                    </span>
                    <Card.Link href="#" className={styles.cardLink}>
                      <i className="bi bi-arrow-right-circle" />
                    </Card.Link>
                  </Card.Body>
                </Card>
              );
            })}
          </ul>
        </>
      )}
    </Container>
  );
};

DetailsCompany.propTypes = {
  company: PropTypes.shape(PropTypes.string).isRequired,
  profile: PropTypes.shape(PropTypes.string).isRequired,
};

export default DetailsCompany;
