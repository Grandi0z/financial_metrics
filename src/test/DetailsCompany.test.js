import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { configureStore } from '@reduxjs/toolkit';
import '@testing-library/jest-dom/extend-expect';
import {
  MemoryRouter, Route, Routes,
} from 'react-router-dom';
import DetailsCompany from '../components/DetailsCompany';
import companiesReducer from '../redux/features/mainPage/companiesSlice';
import companyDetailsReducer from '../redux/features/companyDetails/companyDetailsSlice';

const store = configureStore({
  reducer: {
    companies: companiesReducer,
    companyDetails: companyDetailsReducer,
  },
});

describe('Main Page', () => {
  it('Should match the snapshot', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <MemoryRouter initialEntries={['/details']}>
          <Routes>
            <Route
              path="/details"
              element={(
                <DetailsCompany
                  profile={{
                    companyName: 'company',
                    fullTimeEmployees: '4',
                    image: 'image.jpg',
                  }}
                />
)}
            />
          </Routes>
        </MemoryRouter>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
