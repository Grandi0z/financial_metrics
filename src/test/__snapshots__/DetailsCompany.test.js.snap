// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Page Should match the snapshot 1`] = `
<div
  className="container"
  style={
    Object {
      "padding": 0,
    }
  }
/>
`;
